src/backend/main/go.backends/index/elasticsearch/messages.go

```diff
@@ -174,6 +175,24 @@ func (es *ElasticSearchBackend) GetMessagesRange(filter objects.IndexSearch) (me
 	return messages, totalFound, nil
 }
 
+// return false if message not found or error
+func (es *ElasticSearchBackend) SeekMessageByExternalRef(userID, externalMessageID string) (bool, error) {
```

Comment:
So an elasticsearch lookup for each message to check if it was previously received, I don't like it.

It should be best to create a lookup table in cassandra to map user_id, external_ref -> message_id

Hence no use of shard_id to query the index, is not ok

Reply:
OK. I close PR to make use of cassandra instead of elasticsearch and add a new lookup table : `message_externalref_lookup`